keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -1235,374 +1267,326 @@ const (
 
-	// DeallocationRequested gets set by the scheduler when it detects
-	// the situation where pod scheduling cannot proceed because some
-	// claim was allocated for a node that cannot provide some other
-	// required resource.
-	//
-	// The driver then needs to deallocate this claim and the scheduler
-	// will try again.
-	//
-	// While DeallocationRequested is set, no new users may be added
-	// to ReservedFor.
-	DeallocationRequested bool
-
 	// ReservedFor indicates which entities are currently allowed to use
-	// the resource.  Usually those are Pods, but other objects are
-	// also possible as long as they exist. The resource controller will
-	// remove all entries for objects that do not exist.
+	// the claim. A Pod which references a ResourceClaim which is not
+	// reserved for that Pod will not be started.
 	//
-	// A scheduler must add a Pod that it is scheduling. This must be done
-	// in an atomic ResourceClaim update because there might be multiple
-	// schedulers working on different Pods that compete for access to the
-	// same ResourceClaim, the ResourceClaim might have been marked
-	// for deletion, or even been deallocated already.
+	// There can be at most 32 such reservations. This may get increased in
+	// the future, but not reduced.
+	// +optional
+	ReservedFor []ResourceClaimConsumerReference
+
+	// DeallocationRequested indicates that a ResourceClaim is to be
+	// deallocated.
 	//
-	// kubelet will check this before allowing a Pod to run because a
-	// a user might have selected a node manually without reserving
-	// resources or a
-	// scheduler might have missed that step, for example because it
-	// doesn't support dynamic resource allocation or the feature was
-	// disabled.
+	// The driver then must deallocate this claim and reset the field
+	// together with clearing the Allocation field.
 	//
-	// The maximum size is 32 (= [ResourceClaimReservedForMaxSize]).
-	// This is an artificial limit to prevent
-	// a completely unbounded field in the API.
-	ReservedFor []ResourceClaimUserReference
-
-	<<[UNRESOLVED pohly]>>
-	We will have to discuss use cases and real resource drivers that
-	support sharing before deciding on a) which limit is useful and
-	b) whether we need a different API that supports an unlimited
-	number of users.
-
-	Any solution that handles reservations differently will have to
-	be very careful about race conditions.
-	<<[/UNRESOLVED]>>
+	// While DeallocationRequested is set, no new consumers may be added to
+	// ReservedFor.
+	// +optional
+	DeallocationRequested bool
 }
 
-// ReservedForMaxSize is maximum number of entries in
-// [ResourceClaimStatus.ReservedFor].
+// ReservedForMaxSize is the maximum number of entries in
+// claim.status.reservedFor.
 const ResourceClaimReservedForMaxSize = 32
+```
 
+DeallocationRequested gets set by the scheduler when it detects
+the situation where pod scheduling cannot proceed because some
+claim was allocated for a node that cannot provide some other
```

Comment:
What do you mean by "some other resource". Should this be: "... cannot provide the resource it refers to".

Reply:
"that" refers to the node.

".. for a node for which some other pending claims cannot be allocated
because that node ran out of resources for those." ?

Reply:
yeah, that is clearer.